mount/mount.go

```diff
@@ -22,3 +46,39 @@ func All(mounts []Mount, target string) error {
 	}
 	return nil
 }
+
+// TempMount mounts the provided mounts to a temp dir, and pass the temp dir to f.
+// Finally we will unmount and remove the temp dir regardless of the result of f.
+func TempMount(mounts []Mount, m Mounter, f func(root string) error) error {
+	if len(mounts) == 0 {
+		return errdefs.ErrInvalidArgument
+	}
+	if m == nil {
+		m = mounter{}
+	}
+
+	root, err := ioutil.TempDir("", "containerd-TempMount")
+	if err != nil {
+		return errors.Wrapf(err, "failed to create %s", root)
+	}
+	// We change RemoveAll to Remove so that we either leak a temp dir
+	// if it fails but not RM snapshot data. refer to #1868 #1785
+	defer os.Remove(root)
+
+	// We should do defer first, or we will not do Unmount when only partful Mount fail.
+	defer func() {
+		if uerr := m.Unmount(root); uerr != nil {
```

Comment:
Does an unmount need to be performed for each mount?

Reply:
This really be a problem.:)
The Mounter interface has Unmount method. 
```
// Mounter handles mount and unmount
type Mounter interface {
	Mount(target string, mounts ...Mount) error
	Unmount(target string) error
}
```
I don't know if it is Unmount's responsibility to unmount all from target. 
I think it should be. After all Mount() method mount all.
So I modified  the implementation of mounter.Unmount.
```
func (m mounter) Unmount(target string) error {
	return UnmountAll(target, 0)
}
```

Reply:
done